### DISCUSSION TITLE
Blurry browser workaround

### DISCUSSION THREAD

**Discussion by GerryOnGithub** [cefsharp/CefSharp#4560] [General]
I had some problems with the display of CefSharp being blurry, but nothing I could find here fixed the problem. What I found was if I clicked on the browser window the fonts would clear up, so my fix looks like below. The height stuff is used by my app to resize the window to fit the content "better". The actual fix is the **webBrowser.Focus()**  part.

webBrowser.FrameLoadEnd += BrowserFrameLoadEnd; // register an event handler

```
        private async void BrowserFrameLoadEnd(object? sender, FrameLoadEndEventArgs e)
        {
            if (!e.Frame.IsMain)
                return;
            try
            {
                // cool, run some javascript
                var heightresult = await webBrowser.EvaluateScriptAsync("(function() { return document.body.scrollHeight; })();");
                if (heightresult.Success && heightresult.Result != null)
                {
                    int height = (int)heightresult.Result;
                    Debug.WriteLine("Rendered Height: " + height);
                    mycontainer.SetHeight(webBrowser, height);

                    // run on the main thread
                    Application.Current.Dispatcher.Invoke(() =>
                    {
                        webBrowser.Focus(); 
                    });
                }
            }
            catch(Exception ex)
            {
                Debug.WriteLine("BrowserFrameLoadEnd: " + ex.Message);
            }
        }
```

**Comment by amaitland** [comment]
Does the problem reproduce with the `MinimalExample`?

Please provide more information, see https://github.com/cefsharp/CefSharp/wiki/Questions-&-Support#when-asking-a-question-please-include-the-following

**Reply by GerryOnGithub** [reply]
I just wanted to post my fix for others to try. But I found it it is best to add some code to delay webBrowser.Focus() for "some time". How much time? Not sure, I'm trying around 50ms and that seems to be enough; it might be enough just to give up the thread.

Minimal example works fine for me. I'll look at it some more, see if there is something I'm missing.

Thanks for CefSharp - is a key component in my application
https://imgur.com/a/NmnzW7D

**Reply by amaitland** [reply]
> I just wanted to post my fix for others to try.

Technically this shouldn't be necessary. 

Make your application has an `app.manifest` as per https://github.com/cefsharp/CefSharp/wiki/Quick-Start-For-MS-.Net-4.x#40-add-appmanifest-to-your-application

> Please provide more information, see https://github.com/cefsharp/CefSharp/wiki/Questions-&-Support#when-asking-a-question-please-include-the-following

If you require further assistance then please answer the questions as per the link
